contracts/SnapshotToken/Extensions/SimpleVote.sol

```diff
@@ -0,0 +1,191 @@
+    uint256 public VOTING_PERIOD;
+
+    /////////////////////////
+    // Mutable state
+    ////////////////////////
+    mapping (uint256 => Proposal) private _proposals;
+    mapping (address => bool) private _hasActiveProposal;
+    uint256 private _nextProposalId;
+
+    /////////////////////////
+    // Events
+    ////////////////////////
+    event LogNewProposal(address indexed owner, uint256 indexed proposalId, uint256 snapshotId, uint256 endTime);
+    event LogProposalResult(uint256 indexed proposalId, bool hasPassed, uint256 inFavor, uint256 against);
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    /// @param token which balances are used for voting
+    /// @param votingPeriodInDays amount of days that each vote will take
+    constructor(
+        ISnapshotableToken token,
+        uint256 votingPeriodInDays
+    )
+        public
+    {
+        VOTING_PERIOD = votingPeriodInDays * 1 days;
+        TOKEN = token;
+    }
+
+    /////////////////////////
+    // Public functions
+    ////////////////////////
+
+    /// @dev Creates a proposal, uniquely identifiable by its assigned proposalId
+    function addProposal() public {
+        require(!_hasActiveProposal[msg.sender], "Only one active proposal per address");
+
+        uint256 sId = TOKEN.currentSnapshotId();
+        // advance snapShotID, so that any subsequent trades will be in the non-finalized snapshot
+        TOKEN.createSnapshot();
+
+        _proposals[_nextProposalId++] = Proposal({
+            inFavor: 0,
+            against:0,
+            endTime: now + VOTING_PERIOD,
+            owner: msg.sender,
+            snapshotId: sId
+        });
+
+        _hasActiveProposal[msg.sender] = true;
+
+        emit LogNewProposal(msg.sender, _nextProposalId - 1, sId, now + VOTING_PERIOD);
+    }
+
+    /// @dev increase the votecount on a given proposal by the token balance of the sender
+    ///   throws if proposal does not exist or the vote on it has ended already. Votes are final,
+    ///   changing the vote is not allowed
+    /// @param proposalId of the proposal to be voted on
+    /// @param voteInFavor of the desired proposal
+    function vote(uint256 proposalId, bool voteInFavor) public {
+        incrementVote(proposalId, voteInFavor, msg.sender);
+    }
+
+    /// @dev same as vote, only for a relayed vote. Will throw if provided signature (v,r,s) does not match
+    ///  the address of the voter
+    /// @param voter address whose token balance should be used as voting weight
+    function relayedVote(
+        uint256 proposalId,
+        bool voteInFavor,
+        address voter,
+        bytes32 r,
+        bytes32 s,
+        uint8 v
+    )
+        public
+    {
+        // check that message signature matches the voter address
+        // solium-disable indentation
+        require(ecrecover(
+            keccak256(abi.encodePacked(
+                "\x19Ethereum Signed Message:\n32",
```

Comment:
we could probably optimize this part by pre-encoding `"\x19Ethereum Signed Message:\n32", keccak256(abi.encodePacked(byte(0)` but it would (probably) involve some inline assembly trickery. 
do you know if this `keccak256(abi.encodePacked(byte(0)` will be evaluated at compile time?